src/runtime/channel.h

```diff
@@ -17,14 +17,34 @@ extern "C"
     {
         struct lock lock;
         struct condition_variable notify_space_available;
+
+        /// Pointer to the start of the channel's buffer.
         uint8_t* data;
-        size_t capacity, head, high, cycle, mapped;
+
+        /// Maximum number of bytes this channel can hold.
+        size_t capacity;
+
+        /// Position in the channel where the next write operation will occur.
+        size_t head;
+
+        /// Highest position in the channel that has been written to in the current cycle.
+        size_t high;
+
+        /// Number of times the buffer has been filled and wrapped around to the start.
+        size_t cycle;
+
+        /// Pointer to end position of reserved region before committing a write.
+        size_t mapped;
```

Comment:
@nclack wording here is awkward, but the best I can think of. If you've got a better way to put it, feel free to commit it.

Reply:
Positions on the queue are described by both an offset in the buffer and the cycle count. I think of `head` and `cycle` as part of a tuple defining a particular position.

So, those fields might be best logically grouped.

But I think these descriptions are fine!